File: Solutions/Go/euler-7.go
package main

import (
	"fmt"
	"math"
)

func isPrime(n int) bool {
	if n < 2 {
		return false
	}

	if n == 2 || n == 3 {
		return true
	}

	if n%2 == 0 || n%3 == 0 {
		return false
	}

	limit := math.Floor(math.Sqrt(float64(n))) + 2

	for i := 6; i <= int(limit); i += 6 {
		if n%(i-1) == 0 || n%(i+1) == 0 {
			return false
		}
	}

	return true
}

func main() {
	seen := 0
	n := 1

	for seen < 10001 {
		n += 1
		if isPrime(n) {
			seen += 1
		}
	}

	fmt.Println(n)
}

File: Solutions/Python/euler 7.py
from math import sqrt, ceil

def nthPrime(n):
    primes = [2]
    i=1
    while len(primes) < n:
        current = primes[-1] + i
        isPrime = True
        for x in range(2,ceil(sqrt(current)+1)):
            if current % x == 0:
                isPrime = False
        if isPrime == True:
            primes += current,
            i=1
        else:
            i+=1
    return primes[-1]
            
   
print(nthPrime(10001))
